### PR TITLE
Express notes api

### PR DESCRIPTION
**Pull request by samanthavivonia**
image
image
image
image

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  var notesArray = [];
+  for (var property in noteData.notes) {
+    notesArray.push(noteData.notes[property]);
+  }
+  res.json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (req.params.id < 0 || isNaN(req.params.id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:24]: If `id` is a decimal number (like `14.3225`), the response should still be a `400 Bad Request`.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  } else if (!noteData.notes[req.params.id]) {
+    res.status(404).json({ error: 'cannot find note with id ' + req.params.id });
+  } else {
+    res.json(noteData.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body) {
+    res.status(400).json({ error: 'content is a required field' });
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:34]: If the request has a body, but is still missing a `'content'` field (such as `{ ad: 'Shop at Green Burrito' }`), the response should still be a `400 Bad Request`.

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+app.delete('/api/notes/:id', (req, res) => {
+  if (req.params.id < 0 || isNaN(req.params.id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!noteData.notes[req.params.id]) {
+    res.status(404).json({ error: 'cannot find note with id ' + req.params.id });
+  } else {
+    delete noteData.notes[req.params.id];
+    res.sendStatus(204);
+    fs.writeFile('./data.json', JSON.stringify(noteData, null, 2), () => {
+    });
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:59]: A `204` response shouldn't be sent until after the `data.json` file has been successfully written.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: express-notes-api/index.js (hunk 2 above)
Comment (samanthavivonia) [reply express-notes-api/index.js]: I can't seem to find this one so maybe this is older version. I think its fixed

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**